src/components/application_manager/include/application_manager/message_helper.h

```diff
@@ -88,10 +88,83 @@ class MessageHelper {
   static smart_objects::SmartObjectSPtr CreateHMINotification(
       hmi_apis::FunctionID::eType function_id);
 
+  /**
+   * @brief CreateOnSystemRequestNotificationToMobile creates mobile
+   * OnSystemRequestNotification
+   * @param policy_data pt snapshot data
+   * @param connection_key connection key of application
+   * @param request_type OnSystemRequest request type
+   * @return OnSystemRequest notification smart object
+   */
+  static smart_objects::SmartObjectSPtr
+  CreateOnSystemRequestNotificationToMobile(
+      const std::vector<uint8_t>& policy_data,
+      const uint32_t connection_key,
+      const mobile_apis::RequestType::eType request_type);
+
+  /**
+   * @brief ServiceStatusUpdateNotificationBuilder small utility class used for
+   * more flexible construction of OnServiceUpdateNotification
+   */
+  class ServiceStatusUpdateNotificationBuilder {
```

Comment:
Not all parameters are mandatory in this notification and different use cases may involve different parameters, hence there should be several overloads of such function, or some negative values (e.g. INVALID_ENUM) passed when such function is called. Builder was introduced to make construction of notification object simpler and more readable.

Reply:
That can be accomplished using the default parameters, as is the case with the code snippet I included. Personally I think that the `serviceEvent` parameter should be mandatory, but it might not be worth a proposal revision. In any case, a default value could be set for `event` as well, if desired. I think a builder might be justified for a more complex RPC, but I don't believe that it's necessary here.

Reply:
Removed notification builder in c68887aa422573770ecc526fffcebbc5f517c445